pkg/controller/supportbundlecollection/controller.go

```diff
@@ -169,6 +173,8 @@ func (c *Controller) Run(stopCh <-chan struct{}) {
 
 	go wait.Until(c.worker, time.Second, stopCh)
 
+	go c.StatusController.Run(stopCh)
```

Comment:
This looks a bit strange. If we decouple supportBundleController and statusBundleStatusController, it doesn't make sense to inject the latter to the former and let the former to start both. As seen from the change, the only purpose of this injection is to start the status controller.

But here I feel there is no need to have a separate statusController, as the existing controller would also update status. If we just add some logic to the existing controller, many repeated code could be saved and no need to worry about race conditions between the two controllers.